### PR TITLE
Refractor attention into separate block, support flash attention.

### PR DESCRIPTION
**Pull request by mikecovlee**

### CODE REVIEW

File: mlora.py [hunk 1]
```diff
@@ -42,6 +42,8 @@
                     help="Disable the adapter modules")
 parser.add_argument('--tokenizer', type=str,
                     help='Path to or name of tokenizer')
+parser.add_argument('--flash_attn', action='store_true',
```
Comment (yezhengmao1) [inline mlora.py]: from "clean code" by Robert C. Martin, it's suggest don't use flag argument. split or use type to replace it.

Comment (mikecovlee) [reply mlora.py]: How about changing to `--attn_impl auto|eager|xformers|flash_attn`?

Comment (yezhem) [reply mlora.py:45]: use attn_impl maybe clear

Comment (yezhengmao1) [reply mlora.py]: GOOD

**Pull request merged**